src/core/sagas/utils.js

```diff
@@ -1,19 +1,24 @@
 /* @flow */
 import { ErrorHandler } from 'core/errorHandler';
-import log from 'core/logger';
+import defaultLog from 'core/logger';
 import type { ApiStateType } from 'core/reducers/api';
 
-export function createErrorHandler(id: string): typeof ErrorHandler {
+
+export function createErrorHandler(
+  id: string,
+  { log = defaultLog }: {| log: typeof defaultLog |} = {}
```

Comment:
There's nothing readable about that syntax to me. 😢 It looks like information overload and it reads like the annotation isn't an addendum to the code/argument but sandwiched in the middle of the default argument declaration. So I'm reading about what the argument is in the code and "bam!" in the middle is annotation before it reverts to code. I think this is the most frustrating part of flow to me: it isn't an annotation as much as it is an interruption.